### PR TITLE
fix(android): various attachment picker issues

### PR DESCRIPTION
**Pull request by santhoshvai**
## 🎯 Goal

There are many problems with attachment picker on Android that this PR aims to fix:
1. Random popups of the bottom sheet #1283 (need to release this in beta and ask customers to confirm if this is indeed fixed)
2. When gallery permission is denied, the asking for permission error is only shown when the empty bottom sheet is dragged down
3. When gallery permission is denied, "reconnecting..." bar is shown off and removed on and on.. this is because the app goes background to foreground constantly due to requesting of permissions (which shows the android system dialog over the app even transparently)
4. update bottomsheet lib to latest.. as few bug fixes were done compared to the version we currently use.

## 🛠 Implementation details

NA

## 🎨 UI Changes

NA

## 🧪 Testing

- open/close keyboard frequently on android or change android keyboard.. like gif to normal and back
	- bottom sheet may open for a short time but it should stay closed after.. short open is okay due to internal container layout of bottom sheet
- tap message input wait until keyboard opens and then open attachment picker
	- attachment picker should stay open and keyboard should close
- open attachment picker, wait till it opens. Then tap the message input
	- attachment picker should close and the keyboard should open

## ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [ ] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [ ] Expo iOS and Android

### CODE REVIEW

**Comment by vanGalilea** [comment]
installation & build package failed ```sampleapp: error An unexpected error occurred: "ENOENT: no such file or directory, lstat '/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/native-package/node_modules'".```

**Comment by santhoshvai** [comment]
> installation & build package failed `sampleapp: error An unexpected error occurred: "ENOENT: no such file or directory, lstat '/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/native-package/node_modules'".`

@vanGalilea this was a bug in the `check_pr` GH workflow. It did not build the native package.

**Comment by khushal87** [comment]
Screenshot_1666072149
This happens for like a fraction of second

**Comment by khushal87** [comment]
Opening and closing the attachment picker and keyboard repeatedly on Android triggers the Reconnecting label which stays on the screen for a long time when the permission was denied initially. Is it a good behaviour?

**Comment by khushal87** [comment]
Overall good work with the PR. Thanks for adding the description and test steps. 😄

File: package/src/components/MessageList/NetworkDownIndicator.tsx [hunk 1]
```diff
@@ -33,7 +34,11 @@ export const NetworkDownIndicator = () => {
   } = useTheme();
   const { t } = useTranslationContext();
 
-  if (isOnline && !error) {
+  // when requesting permissions on Android from attachment picker, the app goes to background and foreground in a short time..
+  // to avoid reshowing the network down indicator, we debounce the isOnline value
+  const debouncedIsOnline = useDebouncedValue(isOnline, 1000);
+
+  if (debouncedIsOnline && !error) {
```
[the remark this answers is not in the document]
Comment (santhoshvai) [reply package/src/components/MessageList/NetworkDownIndicator.tsx:41]: No.. it is only debouncing.. meaning a change is only accepted every 1 second.. you dont change the isOnline indicator less than a second

**Comment by santhoshvai** [comment]
> 
> 
> This happens for like a fraction of a second. Not a bug but the feel looks weird. You can avoid this too.

This happens when tapping the message input; after opening the bottom sheet right 

it is unfortunately unavoidable..  the bottom sheet pads the view every time we open the keyboard on android. 
The close event that we call is processed after that.

**Comment by vanGalilea** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=GetStream_stream-chat-react-native&pullRequest=1760)
> 
> [Bug](https://sonarcloud.io/project/issues?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=CODE_SMELL) [C](https://sonarcloud.io/project/issues?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=CODE_SMELL) [25 Code Smells](https://sonarcloud.io/project/issues?id=GetStream_stream-chat-react-native&pullRequest=1760&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=GetStream_stream-chat-react-native&pullRequest=1760&metric=new_coverage&view=list) [53.8% Coverage](https://sonarcloud.io/component_measures?id=GetStream_stream-chat-react-native&pullRequest=1760&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=GetStream_stream-chat-react-native&pullRequest=1760&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=GetStream_stream-chat-react-native&pullRequest=1760&metric=new_duplicated_lines_density&view=list)

25 code smells 😢

**Comment by santhoshvai** [comment]
> Opening and closing the attachment picker and keyboard repeatedly on Android triggers the Reconnecting label which stays on the screen for a long time when the permission was denied initially. Is it a good behaviour?

Slightly better than before. Now the label stays for a second. I wil reduce tge debounce and see if its better

**Review by vanGalilea — APPROVED** [review]

**Review by vanGalilea — APPROVED** [review]

**Pull request merged**